src/main/java/seedu/address/storage/JsonReminderListStorage.java

```diff
@@ -0,0 +1,79 @@
+package seedu.address.storage;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.util.FileUtil;
+import seedu.address.commons.util.JsonUtil;
+import seedu.address.model.reminder.ReadOnlyReminderList;
+
+/**
+ * A class to access ReminderList data stored as a json file on the hard disk.
+ */
+public class JsonReminderListStorage implements ReminderListStorage {
+    private static final Logger logger = LogsCenter.getLogger(JsonReminderListStorage.class);
+
+    private Path filePath;
+
+    public JsonReminderListStorage(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    public Path getReminderListFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public Optional<ReadOnlyReminderList> readReminderList() throws DataConversionException {
+        return readReminderList(filePath);
+    }
+
+    /**
+     * Similar to {@link #readReminderList()}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     * @throws DataConversionException if the file is not in the correct format.
+     */
+    public Optional<ReadOnlyReminderList> readReminderList(Path filePath) throws DataConversionException {
+        requireNonNull(filePath);
+
+        Optional<JsonSerializableReminderList> jsonReminderList = JsonUtil.readJsonFile(
+                filePath, JsonSerializableReminderList.class);
+        if (!jsonReminderList.isPresent()) {
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.of(jsonReminderList.get().toModelType());
+        } catch (IllegalValueException ive) {
```

Comment:
can this be done more succinctly with optional.map?

Reply:
Don't seem like it can be done in the suggested way as there will be some issues with the exceptions thrown.